test/unit/plugins/validate-semantic/2and3/schemas.js

```diff
@@ -169,3 +169,314 @@ describe(`validation plugin - semantic - 2and3 schemas`, () => {
     })
   })
 })
+
+describe(`values in Enum must be instance of the defined type`, () => {
+  // Numbers tests
+  it("should return an error for a text value in a enum number type in OpenApi 3", () => {
+    const spec = {
+      openapi: "3.0.0",
+      "paths": {
+        "/pets": {
+          "get": {
+            "parameters": [
+              {
+                name: "number",
+                in: "query",
+                schema: {
+                  type: "number",
+                  enum: [1, Text, 3]
```

Comment:
```suggestion
                  enum: [1,"text", 3]
```

`Text` is a function since we're in JavaScript here, but at a glance one could misread it as a YAML string that translates to `[1, "Text", 3]` — let's change this for clarity's sake.

Reply:
Changed.